### PR TITLE
Add Plex Live TV channel watching

### PR DESCRIPTION
**Pull request by tmcleroy**
## What

Adds a minimal **Live TV** experience to the Plex module — watch currently-live channels from a server that has a tuner + DVR. No DVR/recording features.

A synthetic **LIVE TV** row appears on the Plex home screen when the active server has a DVR. Selecting it lists the channel lineup; choosing a channel tunes it and plays the live HLS stream through mpv.

## How it works

- **Channels:** `GET /livetv/dvrs` finds the DVR + its EPG lineup, then `GET /livetv/epg/lineupchannels` provides named channels (number + title).
- **Tuning:** `POST /livetv/dvrs/{id}/channels/{channelKey}/tune` allocates a tuner; the playable path comes from the grab operation's `Metadata.key` (`/livetv/sessions/{uuid}`). That path is fed to the universal transcoder for an HLS URL, reusing the existing master-m3u8 parse and `streamUrlReady` → `mpvController.loadAndPlay` hand-off.
- **Tuner lifecycle:** a timeline keep-alive ping while watching, and an explicit transcode-stop on exit so the tuner is released.

## Changes

- `PlexBackend`: `load_live_channels`, `tune_channel`, `update_live_timeline`, `stop_live_session`; a `/livetv/dvrs` probe in `load_libraries_impl` that injects the synthetic home row only when a DVR exists.
- QML: a `live_tv` branch in `Libraries.qml`, a new `LiveChannels.qml` channel list (modeled on `Libraries.qml`), and a new slimmed `LivePlayer.qml`.

## Not included (on purpose)

In-player channel switching is intentionally left out for now — to change channels you exit back to the channel list and pick another. Routing channel-change keys past mpv during fullscreen playback needs a design decision, so that's deferred for discussion.

## Testing notes

Requires a Plex server with a configured tuner + DVR. The live-endpoint response shapes were confirmed against a real server.

Manually human-tested on both targets: a Mac, and a Raspberry Pi 3B+ hooked up to a CRT. On each, the LIVE TV home row appears, channels list, a selected channel tunes and plays, the mpv OSC works during playback, and exiting releases the tuner.

[Demo video](https://drive.google.com/file/d/13C96kSUpa-ZdAiCXqTIzBubEhbyfkoE3/view?usp=sharing)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: modules/plex/views/LivePlayer.qml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    Connections {
+        target: plexBackend
+
+        function onStreamUrlReady(url, plexToken) {
+            livePlayerRoot.streamUrl = url
+            livePlayerRoot.plexToken = plexToken
+            // Live always transcodes (HLS): no separate audio/sub tracks to pick.
+            mpvController.loadAndPlay(url, 0, 0, -1, [], false, -1, 0.0, plexToken)
```
Comment (snkinard) [inline modules/plex/views/LivePlayer.qml]: This comment was generated by Codex using GPT 5.5.

`loadAndPlay` takes `subFiles` and then `subLangs` before the `loop` flag. This call skips the empty `subLangs` argument, so `false`, `-1`, `0.0`, and `plexToken` are shifted into the wrong parameter slots. In QML that either fails method resolution or launches mpv without the Plex token, leaving Live TV unable to start playback after tuning. Please mirror the existing Plex player calls and pass both empty lists, e.g. `loadAndPlay(url, 0, 0, -1, [], [], false, -1, 0.0, plexToken)`.

**Comment by anthonycaccese** [comment]
thank you @tmcleroy, i sadly don’t have a tv tuner set up on my plex server so i won’ be able to test this directly but i’ll do a review of the code sometime this week and add feedback/questions.  very cool addition, thank you as always.

File: modules/plex/views/LivePlayer.qml (hunk 1 above)
Comment (anthonycaccese) [reply modules/plex/views/LivePlayer.qml]: thank you @snkinard

**Comment by snkinard** [comment]
@anthonycaccese @tmcleroy If I get a chance I am going to build this on my pi and test locally. If I'm able to do so I'll post a follow up here.

**Comment by tmcleroy** [comment]
@anthonycaccese let me know if you want me to test anything specifically, I'm happy to record a demo video, I of course don't expect you to set up a TV tuner just for this and I happen to have one set up already. My [demo video](https://drive.google.com/file/d/13C96kSUpa-ZdAiCXqTIzBubEhbyfkoE3/view?usp=sharing) linked in the description was taken at the time of PR creation, I ran it again watching world cup matches for a few hours after the [commit to fix the issue with crashing after a few minutes](https://github.com/anthonycaccese/240-MP/pull/89/changes/5105d879d2fcf63c181b980a0b765469d40750a9), and everything seems stable.

**Comment by snkinard** [comment]
## Plex Live TV channel list stays on `LOADING...`

@tmcleroy : I discovered an error path while attempting to do a happy path hardware test. I was able to select `Live TV`, but then was not authorized to access the Live TV of the server I was connecting to (likely my auth on the local pi needs to be refreshed to pick up new permissions). 
- When this edge case happens, it results in a hung loading screen rather than confirming access is forbidding and backing out

Tested this on a Raspberry Pi / CRT 240-MP setup with AI assistance from Codex.

### Steps

1. Open Plex.
2. Select `LIVE TV`.
3. Wait for the channel list to load.

### Observed behavior

The Live TV screen remains on:

```text
LOADING...
```

for 30+ seconds and never shows channels or an error state.

The runtime log shows the backend receives an error from Plex:

```text
[LiveChannels] Error: LOAD CHANNELS FAILED:
Error transferring ... /livetv/epg/lineupchannels?... - server replied: Forbidden
```

I also ran a token-safe endpoint status probe from the Pi. With the same per-server token:

```text
/livetv/dvrs                   -> 200
/livetv/epg/lineupchannels?... -> 403 Forbidden
```

### Notes

This suggests the Live TV row can appear because `/livetv/dvrs` succeeds, but the channel list depends on `/livetv/epg/lineupchannels`, which this Plex server rejects with `403`.

There also appears to be a UI error-state issue: `LiveChannels.qml` logs `onErrorOccurred`, but does not transition out of the loading state, so the user sees an indefinite `LOADING...` screen.

**Comment by snkinard** [comment]
> ## Plex Live TV channel list stays on `LOADING...`
> @tmcleroy : I discovered an error path while attempting to do a happy path hardware test. I was able to select `Live TV`, but then was not authorized to access the Live TV of the server I was connecting to (likely my auth on the local pi needs to be refreshed to pick up new permissions).
> 
> * When this edge case happens, it results in a hung loading screen rather than confirming access is forbidding and backing out
> 
> Tested this on a Raspberry Pi / CRT 240-MP setup with AI assistance from Codex.
> 
> ### Steps
> 1. Open Plex.
> 2. Select `LIVE TV`.
> 3. Wait for the channel list to load.
> 
> ### Observed behavior
> The Live TV screen remains on:
> 
> ```
> LOADING...
> ```
> 
> for 30+ seconds and never shows channels or an error state.
> 
> The runtime log shows the backend receives an error from Plex:
> 
> ```
> [LiveChannels] Error: LOAD CHANNELS FAILED:
> Error transferring ... /livetv/epg/lineupchannels?... - server replied: Forbidden
> ```
> 
> I also ran a token-safe endpoint status probe from the Pi. With the same per-server token:
> 
> ```
> /livetv/dvrs                   -> 200
> /livetv/epg/lineupchannels?... -> 403 Forbidden
> ```
> 
> ### Notes
> This suggests the Live TV row can appear because `/livetv/dvrs` succeeds, but the channel list depends on `/livetv/epg/lineupchannels`, which this Plex server rejects with `403`.
> 
> There also appears to be a UI error-state issue: `LiveChannels.qml` logs `onErrorOccurred`, but does not transition out of the loading state, so the user sees an indefinite `LOADING...` screen.

1) confirmed that the same user (n00bosaurus) is able to access Live TV on the server in question (Gaming PC) via the plex webapp
2) completely refreshed auth on 240mp by logging out and then signing back in
3) Same issue still encountered

Notes from codex:
```
Still reproduces after the full logout/login.

The live Pi log shows the same failure path:

`LiveChannels` requests `/livetv/epg/lineupchannels?...` and Plex replies `Forbidden`, so the UI stays on `LOADING...`.

That means the fresh auth did not resolve it. The behavior is consistent with what we saw before: Live TV is visible enough to enter the screen, but the channel lineup endpoint used by the PR is still denied for this account/server/token shape on your Pi setup.
```

**Comment by snkinard** [comment]
Hardware test update for the latest PR head `c2efcde7e5c70d7cf4bc59f1cd4e6f45142ba47c`, tested on the Raspberry Pi/CRT setup with AI assistance.

Good news: the Live TV channel list now loads successfully in this setup. That is an improvement over the earlier `lineupchannels` 403/loading behavior.

I then tested channel playback three ways:

1. Shared/home Plex user, while the Plex web app was already tuned to a channel on the same server.
2. Same shared/home Plex user after disconnecting the Plex web app from the tuner.
3. Plex user that owns the server hosting the tuner.

In all three cases, selecting `Fox 7.1` moved to the player screen and displayed `Tuning 7.1 fox`, but playback did not start and controls stopped responding. The logs show that the backend does obtain a Live TV stream URL each time, for example:

```text
[Plex] Live stream URL for mpv: "https://.../video/:/transcode/universal/session/.../base/index.m3u8"
```

Immediately after that, QML fails before `mpv` is launched:

```text
"Could not convert argument 5 at"
         "onStreamUrlReady@file:///home/snkinard/dev/240-MP/modules/plex/views/LivePlayer.qml:91"
file:///home/snkinard/dev/240-MP/modules/plex/views/LivePlayer.qml:91: TypeError: Passing incompatible arguments to C++ functions from JavaScript is not allowed.
```

Process inspection during the hang showed `240mp` running but no `mpv` process, which matches the log path: the stream URL is created, but playback handoff does not happen.

One additional edge case to consider from test 1: if no tuner is available on the host server, the UI should probably fail safely instead of leaving the user stuck. In this specific follow-up, though, freeing the tuner and trying the server-owner account both reproduced the same `LivePlayer.qml:91` argument-conversion failure, so that appears to be the current playback blocker.

**Comment by snkinard** [comment]
> Hardware test update for the latest PR head `c2efcde7e5c70d7cf4bc59f1cd4e6f45142ba47c`, tested on the Raspberry Pi/CRT setup with AI assistance.
> 
> Good news: the Live TV channel list now loads successfully in this setup. That is an improvement over the earlier `lineupchannels` 403/loading behavior.
> 
> I then tested channel playback three ways:
> 
> 1. Shared/home Plex user, while the Plex web app was already tuned to a channel on the same server.
> 2. Same shared/home Plex user after disconnecting the Plex web app from the tuner.
> 3. Plex user that owns the server hosting the tuner.
> 
> In all three cases, selecting `Fox 7.1` moved to the player screen and displayed `Tuning 7.1 fox`, but playback did not start and controls stopped responding. The logs show that the backend does obtain a Live TV stream URL each time, for example:
> 
> ```
> [Plex] Live stream URL for mpv: "https://.../video/:/transcode/universal/session/.../base/index.m3u8"
> ```
> 
> Immediately after that, QML fails before `mpv` is launched:
> 
> ```
> "Could not convert argument 5 at"
>          "onStreamUrlReady@file:///home/snkinard/dev/240-MP/modules/plex/views/LivePlayer.qml:91"
> file:///home/snkinard/dev/240-MP/modules/plex/views/LivePlayer.qml:91: TypeError: Passing incompatible arguments to C++ functions from JavaScript is not allowed.
> ```
> 
> Process inspection during the hang showed `240mp` running but no `mpv` process, which matches the log path: the stream URL is created, but playback handoff does not happen.
> 
> One additional edge case to consider from test 1: if no tuner is available on the host server, the UI should probably fail safely instead of leaving the user stuck. In this specific follow-up, though, freeing the tuner and trying the server-owner account both reproduced the same `LivePlayer.qml:91` argument-conversion failure, so that appears to be the current playback blocker.

One final important note. I tried one more time while observing the dashboard for the owner's server. Interestingly, when 240mp attempts to tune fox 7.1 and hangs, the dashboard shows that user receiving a stream for the same live TV station:
240mp_PR89_plex_dashboard

**Comment by snkinard** [comment]
Hardware test update on latest PR head `c5833a14ffe32e7e0c60f0a1ce38f3c74ce788d1`, tested with Codex assistance and manual human verification on the Raspberry Pi / CRT setup.

**RESOLVED:** the prior Live TV playback handoff failure at `LivePlayer.qml:91` is fixed on this head.

Validation performed:

1. Pulled the latest exact PR head, without merging or rebasing it with any other code.
2. Confirmed the `LivePlayer.qml` call now passes the missing `subLangs` argument to `mpvController.loadAndPlay(...)`.
3. Ran `git diff --check` successfully.
4. Built successfully on macOS.
5. Synced and built successfully on the Pi.
6. Launched the dev build on the CRT Pi setup.
7. Manually tested Plex -> Live TV -> channel playback.

Observed result:

- Live TV channel list loads.
- Selecting a Live TV channel now starts playback successfully.
- The previous QML error no longer appears:

```text
LivePlayer.qml:91: TypeError: Passing incompatible arguments to C++ functions from JavaScript is not allowed.
```

Runtime evidence from the Pi shows `mpv` running against the Plex Live TV HLS stream, and the mpv log reports active AV playback/progress. This confirms the stream URL is now handed off to playback instead of getting stuck on the `Tuning...` screen.

**Comment by snkinard** [comment]
image

**Comment by tmcleroy** [comment]
@anthonycaccese Sorry for the comment spam. FYI @snkinard is an IRL friend of mine. He brought his pi over to my house today and we were debugging this together. In its current state the PR works for the server owner user as well as the home group member user (adding user to home group is the only way to grant live tv access). Let us know if there are any other edge cases you'd like us to test.

**Comment by anthonycaccese** [comment]
No apologies needed ever @tmcleroy, this is amazing i love everything about this... friends building together, pictures of commodore 1701 monitors, live tv on a crt again 🤩 

I apologize for my delay getting time to look things over but def feeling good based on your conversation here.  I'll get more time with this tonight but I think you've both answered most (if not all of my questions) so if all looks good I'll aim to merge it. 

Anything else from your end that'd you want to tackle before that?

**Comment by snkinard** [comment]
> No apologies needed ever @tmcleroy, this is amazing i love everything about this... friends building together, pictures of commodore 1701 monitors, live tv on a crt again 🤩
> 
> I apologize for my delay getting time to look things over but def feeling good based on your conversation here. I'll get more time with this tonight but I think you've both answered most (if not all of my questions) so if all looks good I'll aim to merge it.
> 
> Anything else from your end that'd you want to tackle before that?

"Anything else from your end that'd you want to tackle before that?" @anthonycaccese thanks for checking. I spoke with @tmcleroy and we are both aligned that critical path use cases are fully covered and this is ready for your evaluation.

**Comment by anthonycaccese** [comment]
things look good

re: "In-player channel switching" i am def down to talk through design ideas for that in a future PR.  

Will call this good for now to merge.
